testutils/go.mod

```diff
@@ -1,88 +1,87 @@
 module github.com/openshift/cluster-api-actuator-pkg/testutils
 
-go 1.22.1
-
-toolchain go1.22.7
+go 1.22.11
 
 require (
 	github.com/go-logr/logr v1.4.2
-	github.com/golangci/golangci-lint v1.61.0
-	github.com/onsi/ginkgo/v2 v2.22.0
-	github.com/onsi/gomega v1.36.0
+	github.com/golangci/golangci-lint v1.63.4
+	github.com/onsi/ginkgo/v2 v2.22.2
+	github.com/onsi/gomega v1.36.2
 	github.com/openshift/api v0.0.0-20250106182855-361e35fd82e5
-	k8s.io/api v0.31.1
-	k8s.io/apimachinery v0.31.1
-	k8s.io/client-go v0.31.0
-	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
-	sigs.k8s.io/cluster-api v1.8.5
-	sigs.k8s.io/cluster-api-provider-aws/v2 v2.6.1
+	k8s.io/api v0.31.3
+	k8s.io/apimachinery v0.31.3
+	k8s.io/client-go v0.31.3
+	k8s.io/utils v0.0.0-20241210054802-24370beab758
```

Comment:
Does the code compile if these are moved to 0.32.1?

Reply:
For the testutil submodule I don't think we can go up to 0.32, as that would force:
```
go: k8s.io/api@v0.32.1 requires go >= 1.23.0; switching to go1.23.6
```

Which then would force the cluster-capi-operator to be on 1.23, which we've always kept in line with the CAPI providers (they will be on 1.22).

Reply:
4.19 will be built with Go 1.23

I don't think that means that the go.mod minimum version needs to be Go 1.23, but I certainly don't think moving the minimum go.mod version here matters, we don't import this into any of the CAPI repos do we?

Reply:
> 4.19 will be built with Go 1.23

Correct

> we don't import this into any of the CAPI repos do we?
We do in the cluster-capi-operator. I've always tried to keep the minimum version in sync with the CAPI ecosystem. That said it is probably not a hard requirement.

Reply:
I don't think that's going to be a hard requirement, lets not keep that restriction if possible, I don't think that's going to cause us issues

Reply:
Made the switch to 1.23 also for testutils.